src/goto-programs/remove_exceptions.cpp

```diff
@@ -320,11 +320,17 @@ void remove_exceptionst::add_exception_dispatch_sequence(
     symbol_table.lookup_ref(id2string(function_id)+EXC_SUFFIX);
   symbol_exprt exc_thrown=exc_symbol.symbol_expr();
 
-  // add GOTOs implementing the dynamic dispatch of the
-  // exception handlers
+  std::size_t default_try=0;
+  std::size_t default_catch=(stack_catch.size()>0) ? stack_catch[0].size() : 0;
+
+  // Find the innermost universal exception handler
+  // (with exception type any). We record this one
+  // because no handler after it can possibly catch.
   for(std::size_t i=stack_catch.size(); i-->0;)
```

Comment:
Also, I know not your code but `i-->0` is so counter intutitive could you change to `i>=0;--i)` while you're in the area.